### ISSUE TITLE
Worker process pool only restart one at a time

### ISSUE THREAD

**Issue by lebinh** [celery/celery#2075]
Using the method described at http://docs.celeryproject.org/en/latest/userguide/workers.html#pool-restart-command only restart 1 process in the worker's process pool but not all. The other processes *may* restart  only after the worker received other tasks. How can I make sure all the processes of the worker are restarted?

For example, using tasks defined in this gist https://gist.github.com/lebinh/165f7edb95daa08bee54 and start a worker with concurrency 2: `celery -A tasks worker -l info -c 2`. After sending `pool_restart` command with `tasks.app.control.broadcast('pool_restart')` I got this log:

```
[2014-06-05 18:53:43,310: WARNING/Worker-1] worker process shutdown
[2014-06-05 18:53:44,350: WARNING/Worker-3] worker process init
^C
worker: Hitting Ctrl+C again will terminate all running tasks!

worker: Warm shutdown (MainProcess)
[2014-06-05 18:54:05,368: WARNING/Worker-2] worker process shutdown
[2014-06-05 18:54:05,368: WARNING/Worker-3] worker process shutdown
```

The worker process 1 is restarted but worker process 2 is still there until I terminate the worker.

I'm using the latest version of Celery, 3.1.11 (Cipater), on Ubuntu 14.04 running on EC2.

**Comment by ask** [comment]
Duplicate of #1658

**Comment by sanexodus** [comment]
Celery version 5.3.1 this bug still exists.
